src/Header/Header.tsx

```diff
@@ -24,63 +24,138 @@ const Header = () => {
     }
   };
 
-  const linkTextStyle = {
-    visibility: menu ? 'visible' : '',
-    opacity: menu ? '1' : '',
-    transition: 'opacity 200ms',
-    whiteSpace: 'nowrap',
-  } as React.CSSProperties;
+
+
+const width = menu ? '14rem' : '3rem';
+const StyledNav = styled.nav`
```

Comment:
`width` should be passed in as a prop and maybe we should rename it to something like `menuOpen` or `isMenuOpen`.

It would look something like....
```jsx
interface StyledNavProps { menuOpen: boolean }
const StyledNav = styled.nav`
  // ...
  width: ${({ menuOpen }): StyledNavProps} => menuOpen ? '14rem' : '3rem';
`;